archive/imo/imo1975_q1.lean

```diff
@@ -0,0 +1,49 @@
+/-
+Copyright (c) 2022 Mantas Bakšys. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mantas Bakšys
+-/
+import data.real.basic
+import data.nat.interval
+import algebra.order.rearrangement
+import algebra.big_operators.ring
+
+/-!
+# IMO 1975 Q1
+
+Let `x₁, x₂, ... , xₙ` and `y₁, y₂, ... , yₙ` be two sequences of real numbers, such that
+`x₁ ≥ x₂ ≥ ... ≥ xₙ` and `y₁ ≥ y₂ ≥ ... ≥ yₙ`. Prove that if `z₁, z₂, ... , zₙ` is any permutation
+of `y₁, y₂, ... , yₙ`, then
+`∑ i in finset.Icc 1 n, (x i - y i) ^ 2 ≤ ∑ i in finset.Icc 1 n, (x i - z i) ^ 2`
+
+# Solution
+
+Firstly, we expand the squares withing both sums and distribute into separate finite sums. Then,
+noting that `∑ i in finset.Icc 1 n, (y i) ^ 2 = ∑ i in finset.Icc 1 n, (z i) ^ 2`, it remains to
+prove that `∑ i in finset.Icc 1 n, x i * z i ≤ ∑ i in finset.Icc 1 n, x i * y i`, which is true by
+the Rearrangement Inequality
+-/
+
+open_locale big_operators
+
+/-- Let `n` be a natural number, `x` and `y` be as in the problem statement and `σ` be the
```

Comment:
```suggestion
/- Let `n` be a natural number, `x` and `y` be as in the problem statement and `σ` be the
```
Doc strings don't apply to variable declarations. I'm not sure why this was accepted, it's probably getting applied as the doc string for the theorem below?